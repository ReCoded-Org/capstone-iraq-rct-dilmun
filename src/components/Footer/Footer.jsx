import React from 'react'
import { useTranslation } from 'react-i18next'
import { NavLink } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  HOME_ROUTE,
  PROFILE_ROUTE,
  CONTACT_ROUTE,
  ABOUT_ROUTE,
} from '../../router'
import Logo from '../../logo.svg'

export default function Footer() {
  const { t } = useTranslation()

  return (
    <footer className="text-white body-font  bg-darkBlue text-center">
      <div className="container px-5 py-24 mx-auto flex md:items-center lg:items-start md:flex-row md:flex-nowrap flex-wrap flex-col">
        <div className="w-54 flex-shrink-0 md:mx-0 mx-auto  md:text-left">
          <a href="/" className="flex  mx-6  ">
            <img className="lg:w-auto md:w-1/2  " src={Logo} alt="logo" />
          </a>
          <h3 className="text-2xl font-medium">Dilmun</h3>
        </div>
        <div className="flex-grow flex flex-wrap md:pl-10 -mb-10 md:mt-0 mt-6 md:text-center">
          <div className="lg:w-1/6 md:w-1/2 w-full px-4 ">
            <h2 className="font-medium tracking-widest text-xl mb-3  border-b-2 ">
              {t('footer.menu')}
            </h2>
            <nav className="list-none mb-10">
              <li>
                <NavLink className=" hover:text-darkgray" exact to={HOME_ROUTE}>
                  {t('navbar.home')}{' '}
                </NavLink>
              </li>
              <li>
                <NavLink className=" hover:text-darkgray" exact to={CONTACT_ROUTE}>
                  {t('navbar.contact')}{' '}
                </NavLink>
              </li>
              <li>
                <NavLink className=" hover:text-darkgray" exact to={PROFILE_ROUTE}>
                  {t('footer.profile')}{' '}
                </NavLink>
              </li>
              <li>
                <NavLink className=" hover:text-darkgray" exact to={ABOUT_ROUTE}>
                  {t('navbar.about')}{' '}
                </NavLink>
              </li>
            </nav>
          </div>
          <div className="lg:w-1/4 md:w-1/2 w-full px-4">
            <h2 className="font-medium text-white-900 tracking-widest text-xl mb-3 border-b-2">
              {t('footer.type')}
            </h2>
            <nav className="list-none mb-10">
              <li>
                <a href='/searchresult?word=&select=Crafted' className="hover:text-darkgray">
                  {t('footer.crafted')}
                </a>
              </li>
              <li>
                <a href='/searchresult?word=&select=Used' className="hover:text-darkgray">
                  {t('footer.used')}
                </a>
              </li>
              <li>
                <a href='/searchresult?word=&select=Donated' className="hover:text-darkgray">
                  {t('footer.donated')}
                </a>
              </li>
            </nav>
          </div>
          <div className="lg:w-1/4 md:w-1/2 w-full ">
            <h2 className="font-medium  tracking-widest text-xl mb-3 border-b-2">
              {t('navbar.contact')}
            </h2>
            <nav className="list-none mb-10">
              <li>
                <div className="flex-shrink item-center justify-center ">
                  <div>
                    <a href="tel:+234081-1236-4568">
                      <FontAwesomeIcon
                        icon="phone"
                        className="hover:text-darkgray "
                      />
                      <span>+234 081-1236-4568</span>
                    </a>
                  </div>
                  <div>
                    <a href="mailto:contact@Dilmun.com">
                      <FontAwesomeIcon
                        icon="envelope"
                        className="hover:text-darkgray "
                      />{' '}
                      <span>contact@Dilmun.com</span>
                    </a>
                  </div>
                </div>
              </li>
            </nav>
          </div>
          <div className="lg:w-1/4 md:w-1/2 w-full px-4">
            <h2 className="font-medium  tracking-widest text-xl mb-3  border-b-2">
              {t('footer.follow')}
            </h2>
            <nav className="list-none mb-10">
              <li>
                <div className="inline-flex sm:ml-auto sm:mt-0 mt-2 justify-center sm:justify-start ">
                  <a href="https://www.instagram.com/">
                    <FontAwesomeIcon
                      icon={['fab', 'instagram']}
                      className="hover:text-darkgray fa-2x mx-2"
                    />
                  </a>
                  <a href="https://www.facebook.com/">
                    <FontAwesomeIcon
                      icon={['fab', 'facebook-square']}
                      className="hover:text-darkgray fa-2x mx-2"
                    />
                  </a>
                </div>
              </li>
            </nav>
          </div>
        </div>
      </div>
      <div className="border-t-2">
        <div className="container mx-auto py-4 px-5 flex flex-wrap flex-col">
          <p className="text-sm "> Â© Copyright 2021 - Dilmun</p>
        </div>
      </div>
    </footer>
  )
}
