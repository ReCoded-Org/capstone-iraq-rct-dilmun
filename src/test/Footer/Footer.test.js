import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import Footer from '../../components/Footer'

it('Footer renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        {' '}
        <Footer />{' '}
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
