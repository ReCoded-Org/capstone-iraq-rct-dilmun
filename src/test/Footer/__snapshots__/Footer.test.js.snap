// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
Array [
  " ",
  <footer
    className="text-white body-font  bg-darkBlue text-center"
  >
    <div
      className="container px-5 py-24 mx-auto flex md:items-center lg:items-start md:flex-row md:flex-nowrap flex-wrap flex-col"
    >
      <div
        className="w-54 flex-shrink-0 md:mx-0 mx-auto  md:text-left"
      >
        <a
          className="flex  mx-6  "
          href="/"
        >
          <img
            alt="logo"
            className="lg:w-auto md:w-1/2  "
            src="logo.svg"
          />
        </a>
        <h3
          className="text-2xl font-medium"
        >
          Dilmun
        </h3>
      </div>
      <div
        className="flex-grow flex flex-wrap md:pl-10 -mb-10 md:mt-0 mt-6 md:text-center"
      >
        <div
          className="lg:w-1/6 md:w-1/2 w-full px-4 "
        >
          <h2
            className="font-medium tracking-widest text-xl mb-3  border-b-2 "
          >
            footer.menu
          </h2>
          <nav
            className="list-none mb-10"
          >
            <li>
              <a
                aria-current="page"
                className=" hover:text-darkgray active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                navbar.home
                 
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className=" hover:text-darkgray"
                href="/contact"
                onClick={[Function]}
              >
                navbar.contact
                 
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className=" hover:text-darkgray"
                href="/profile"
                onClick={[Function]}
              >
                footer.profile
                 
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className=" hover:text-darkgray"
                href="/about"
                onClick={[Function]}
              >
                navbar.about
                 
              </a>
            </li>
          </nav>
        </div>
        <div
          className="lg:w-1/4 md:w-1/2 w-full px-4"
        >
          <h2
            className="font-medium text-white-900 tracking-widest text-xl mb-3 border-b-2"
          >
            footer.type
          </h2>
          <nav
            className="list-none mb-10"
          >
            <li>
              <a
                className="hover:text-darkgray"
                href="/"
              >
                footer.crafted
              </a>
            </li>
            <li>
              <a
                className="hover:text-darkgray"
                href="/"
              >
                footer.used
              </a>
            </li>
            <li>
              <a
                className="hover:text-darkgray"
                href="/"
              >
                footer.donated
              </a>
            </li>
          </nav>
        </div>
        <div
          className="lg:w-1/4 md:w-1/2 w-full px-4"
        >
          <h2
            className="font-medium text-gray-900 tracking-widest text-xl mb-3 border-b-2"
          >
            navbar.contact
          </h2>
          <nav
            className="list-none mb-10"
          >
            <li>
              <div
                className="flex-shrink item-center justify-center "
              >
                <div>
                  <a
                    href="tel:+234081-1236-4568"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-phone fa-w-16 hover:text-darkgray "
                      data-icon="phone"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                    <span>
                      +234 081-1236-4568
                    </span>
                  </a>
                </div>
                <div>
                  <a
                    href="mailto:contact@Dilmun.com"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-envelope fa-w-16 hover:text-darkgray "
                      data-icon="envelope"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                     
                    <span>
                      contact@Dilmun.com
                    </span>
                  </a>
                </div>
              </div>
            </li>
          </nav>
        </div>
        <div
          className="lg:w-1/4 md:w-1/2 w-full px-4"
        >
          <h2
            className="font-medium  tracking-widest text-xl mb-3  border-b-2"
          >
            footer.follow
          </h2>
          <nav
            className="list-none mb-10"
          >
            <li>
              <div
                className="inline-flex sm:ml-auto sm:mt-0 mt-2 justify-center sm:justify-start "
              >
                <a
                  href="https://www.instagram.com/"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-instagram fa-w-14 hover:text-darkgray fa-2x mx-2"
                    data-icon="instagram"
                    data-prefix="fab"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </a>
                <a
                  href="https://www.facebook.com/"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-facebook-square fa-w-14 hover:text-darkgray fa-2x mx-2"
                    data-icon="facebook-square"
                    data-prefix="fab"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </a>
              </div>
            </li>
          </nav>
        </div>
      </div>
    </div>
    <div
      className="border-t-2"
    >
      <div
        className="container mx-auto py-4 px-5 flex flex-wrap flex-col"
      >
        <p
          className="text-sm "
        >
           Â© Copyright 2021 - Dilmun
        </p>
      </div>
    </div>
  </footer>,
  " ",
]
`;
